### ISSUE TITLE
CT Send to Interpreter - breakpoints don't work for records in functions

### ISSUE THREAD

**Issue by ldcouto** [overturetool/overture#438]
If you run combinatorial testing on a VDM-SL model and try to send a test to the interpreter, the test will execute but the interpreter will not suspend on a breakpoint. Likewise, if you violate a pre-condition, etc. the interpreter will not suspend.

Is this a bug? I looked around in the manual and it seems like this should be possible. Without this feature, I think CT loses some of its appeal.

**Comment by ldcouto** [comment]
I've managed to get this to work for some examples that make use of operations but function-based examples definitely have problems.

**Comment by nickbattle** [comment]
This definitely sounds like a bug. You need to be able to debug cases sent to
the interpreter. Does it work as normal for VDM++?

**Comment by ldcouto** [comment]
I think so... but most VDM++ examples use operations. I'll see if I can make a minimal example for the two dialects later today.

BTW, I assume that "Send to Interpreter" always triggers debug mode so any breakpoints are checked.

**Comment by ldcouto** [comment]
Ok, I think I've narrowed it down.

Problem occurs when calling a function with a pre-condition inside an operation. You get a test failure (rather than inconclusive) on the pre-condition failure and sending the test to the interpreter doesn't let you debug it.

You also have to use records, for some reason. The problem happens on both SL and PP models. Here's the examples:

SL:
```
state S of
	x : int

init s == s = mk_S(0)
end;

functions
f : S -> S
f(s) == mk_S(10/s.x)
pre s.x <>0;

operations
opf : S ==> S
opf (s) == f(s);

traces

test:
		opf(S)
```
PP:
```
class A
types
 R :: x:int;

instance variables
	x : R := mk_R(0)

functions
f : R -> R
f(s) == mk_R(s.x)
pre s.x <>0;

operations
opf : R ==> R
opf (s) == f(s);

traces

test:
		opf(x)
		
		
end A
```

![screen shot 2015-03-29 at 19 20 03](https://cloud.githubusercontent.com/assets/582801/6886735/b862cbbe-d648-11e4-8423-25550f4b9356.png)

**Comment by nickbattle** [comment]
Well, it's correct that this isn't inconclusive (that only occurs on the 
"outside" calls that the test makes, since the test may be the problem). But it 
ought to trap into the debugger if there is a fault. Your example is a bit 
strange - you're using the state definition to create a record type which you're 
passing around - but it's legal and should work.

**Comment by ldcouto** [comment]
Yeah... I ran into this while trying to CT a spec generated from a state
machine chart. That's what led to this weird usage pattern.

But the pp version creates a "normal" record and that won't run either
so.... I think there's definitely something up.

**Comment by ldcouto** [comment]
Ok, turns out operations calls are irrelevant. Seems to be mostly about records.

If you try it with the spec below, you can send `demo2`to the interpreter and get breakpoints but you can't for `demo1`.

```
class A

types
 R :: x:int;

instance variables
	x : R := mk_R(0)

functions
f : R -> R
f(s) == mk_R(10/s.x)
pre s.x <>0;

f2 : int -> int
f2(x) == 10/x
pre x <>0

traces
demo:
		f(x)

demo2:
		f2(x.x)		

end A
```

**Comment by nickbattle** [comment]
That's very strange! I suppose we "just" have to debug the debugger :)

**Comment by ldcouto** [comment]
Yeah, that'll be fun...

I would like for @peterwvj  or @lausdahl to weigh in here. They've played around the most with the CT plugin.

**Comment by peterwvj** [comment]
Some of the recent commits seems to have fixed parts of this. Also, I just fixed a small problem with construction of the environment.

Now when I try to execute `demo` and `demo2` it suspends on the pre condition. In addition I tried to change `demo` as shown below.  I can also get it to stop at the `let` as indicated below.

~~~
functions
f : R -> R
f(s) ==
let a = 1, -- BP here
    b = a
in
  mk_R(10/s.x);
~~~

It may be that the verdict is still wrong though (it says failure)
